### PR TITLE
Improve `std::iter::zip` example

### PR DESCRIPTION
**Pull request by andrewbanchich**
`println!` isn't great for doc comments / tests.

### CODE REVIEW

File: library/core/src/iter/adapters/zip.rs [hunk 1]
```diff
@@ -45,15 +45,23 @@ impl<A: Iterator, B: Iterator> Zip<A, B> {
...
 /// let ys = [4, 5, 6];
-/// for (x, y) in zip(&xs, &ys) {
-///     println!("x:{}, y:{}", x, y);
-/// }
+///
+/// let mut iter = zip(xs, ys);
+///
+/// assert_eq!(iter.next().unwrap(), (1,4));
+/// assert_eq!(iter.next().unwrap(), (2,5));
+/// assert_eq!(iter.next().unwrap(), (3,6));
```
Comment (r00ster91) [inline library/core/src/iter/adapters/zip.rs:53]:
```suggestion
/// assert_eq!(iter.next().unwrap(), (1, 4));
/// assert_eq!(iter.next().unwrap(), (2, 5));
/// assert_eq!(iter.next().unwrap(), (3, 6));
```

File: library/core/src/iter/adapters/zip.rs [hunk 2]
```diff
@@ -45,15 +45,23 @@ impl<A: Iterator, B: Iterator> Zip<A, B> {
...
 /// let zs = [7, 8, 9];
-/// for ((x, y), z) in zip(zip(&xs, &ys), &zs) {
-///     println!("x:{}, y:{}, z:{}", x, y, z);
-/// }
+///
+/// let mut iter = zip(zip(xs, ys), zs);
+///
+/// assert_eq!(iter.next().unwrap(), ((1,4), 7));
+/// assert_eq!(iter.next().unwrap(), ((2,5), 8));
+/// assert_eq!(iter.next().unwrap(), ((3,6), 9));
```
Comment (r00ster91) [inline library/core/src/iter/adapters/zip.rs:63]:
```suggestion
/// assert_eq!(iter.next().unwrap(), ((1, 4), 7));
/// assert_eq!(iter.next().unwrap(), ((2, 5), 8));
/// assert_eq!(iter.next().unwrap(), ((3, 6), 9));
```

**Comment by andrewbanchich** [comment]
r? @Mark-Simulacrum 

(not sure if I did that right since I'm not sure what r=me does and Google doesn't help)

**Comment by Mark-Simulacrum** [comment]
Just means that I'll be happy to approve the PR with that change.

@bors r+

**Pull request merged**